disnake/guild.py

```diff
@@ -4567,3 +4580,468 @@ async def create_automod_rule(
+
+    def update_everyone_role(self, *, permissions: Permissions = MISSING) -> PlaceholderID:
+        """
+        Updates attributes of the ``@everyone`` role.
+
+        Parameters
+        ----------
+        permissions: :class:`Permissions`
+            The permissions the role should have.
+
+        Returns
+        -------
+        ``PlaceholderID``
+            The placeholder ID for the ``@everyone`` role.
+            Also available as :attr:`everyone`.
+        """
+        if len(self._roles) == 0:
+            self._roles.append({"id": self._everyone_id})
+        role = self._roles[0]
+
+        if permissions is not MISSING:
+            role["permissions"] = str(permissions.value)
+
+        return self._everyone_id
+
+    def add_role(
+        self,
+        name: str = MISSING,
+        *,
+        permissions: Permissions = MISSING,
+        color: Union[Colour, int] = MISSING,
+        colour: Union[Colour, int] = MISSING,
+        hoist: bool = MISSING,
+        mentionable: bool = MISSING,
+    ) -> PlaceholderID:
+        """
+        Adds a role to the guild builder.
+
+        The default (``@everyone``) role can be referenced using :attr:`everyone`
+        and configured through :func:`update_everyone_role`.
+
+        Parameters
+        ----------
+        name: :class:`str`
+            The role name. Defaults to 'new role'.
+        permissions: :class:`Permissions`
+            The permissions the role should have. Defaults to no permissions.
+        colour: Union[:class:`Colour`, :class:`int`]
+            The colour for the role. Defaults to :meth:`Colour.default`.
+            This is aliased to ``color`` as well.
+        hoist: :class:`bool`
+            Whether the role should be shown separately in the member list.
+            Defaults to ``False``.
+        mentionable: :class:`bool`
+            Whether the role should be mentionable by others.
+            Defaults to ``False``.
+
+        Returns
+        -------
+        ``PlaceholderID``
+            A placeholder ID for the created role.
+        """
+        # always create @everyone role first if not created already
+        if len(self._roles) == 0:
+            self.update_everyone_role()
+
+        _id = self._next_id()
+        data: CreateGuildPlaceholderRole = {"id": _id}
+
+        if name is not MISSING:
+            data["name"] = name
+
+        if permissions is not MISSING:
+            data["permissions"] = str(permissions.value)
+        else:
+            data["permissions"] = "0"
+
+        actual_colour = colour or color or Colour.default()
+        if isinstance(actual_colour, int):
+            data["color"] = actual_colour
+        else:
+            data["color"] = actual_colour.value
+
+        if hoist is not MISSING:
+            data["hoist"] = hoist
+
+        if mentionable is not MISSING:
+            data["mentionable"] = mentionable
+
+        self._roles.append(data)
+        return _id
+
+    def _add_channel(
```

Comment:
Would be nice to move this below _next_id to avoid having private methods between the public ones

Reply:
sounds good.
https://github.com/DisnakeDev/disnake/pull/578/commits/e9021ef55066e609c67ea06f505f5fbe91f4f89f